metadata-service/services/src/main/java/com/linkedin/metadata/service/DataProductService.java

```diff
@@ -58,11 +58,18 @@ public DataProductService(@Nonnull EntityClient entityClient, @Nonnull GraphClie
    * @return the urn of the newly created DataProduct
    */
   public Urn createDataProduct(
-      @Nullable String name, @Nullable String description, @Nonnull Authentication authentication) {
+      @Nullable String id,
+      @Nullable String name,
+      @Nullable String description,
+      @Nonnull Authentication authentication) {
 
     // 1. Generate a unique id for the new DataProduct.
     final DataProductKey key = new DataProductKey();
-    key.setId(UUID.randomUUID().toString());
+    if (id != null && !id.isBlank()) {
+      key.setId(id);
```

Comment:
since we now allow user input for keys here, I believe we should add a check here to ensure that this data product doesn't already exist (just like we do in `CreateDomainResolver.java` for the ID input there)

Reply:
Added error handling